File: crm/src/verticals/healthcare.ts
import { tab } from './shared';
import type { CrmVerticalDefinition } from './types';

export const healthcareVertical: CrmVerticalDefinition = {
  key: 'healthcare',
  label: 'Healthcare',
  riskLevel: 'high',
  entityLabelSingular: 'provider',
  entityLabelPlural: 'providers',
  defaultPlanLabel: 'Healthcare plan',
  clientTabs: [
    tab('overview', 'Overview'),
    tab('readiness', 'Readiness'),
    tab('catalog', 'Providers'),
    tab('appointments', 'Appointments'),
    tab('leads', 'Leads'),
    tab('compliance', 'Compliance'),
    tab('activity', 'Conversations'),
    tab('prompt', 'Prompt'),
    tab('controls', 'Controls'),
  ],
  entityTypes: ['doctor', 'clinic', 'specialty', 'service_line', 'lab_test', 'health_article'],
  readinessChecks: ['providers', 'specialties', 'appointments', 'locations', 'privacy', 'emergency_notice'],
};

File: agent/verticals/healthcare.py
"""Healthcare vertical definition."""

from agent.verticals.base import VerticalDefinition, tabs

VERTICAL = VerticalDefinition(
    key="healthcare",
    label="Healthcare",
    risk_level="high",
    entity_label_singular="provider",
    entity_label_plural="providers",
    default_plan_label="Healthcare plan",
    crm_tabs=tabs(
        ("overview", "Overview"),
        ("readiness", "Readiness"),
        ("catalog", "Providers"),
        ("appointments", "Appointments"),
        ("leads", "Leads"),
        ("compliance", "Compliance"),
        ("activity", "Conversations"),
        ("prompt", "Prompt"),
        ("controls", "Controls"),
    ),
    entity_types=("doctor", "clinic", "specialty", "service_line", "lab_test", "health_article"),
    readiness_checks=("providers", "specialties", "appointments", "locations", "privacy", "emergency_notice"),
    action_types=(
        "SHOW_ENTITIES",
        "SORT_ENTITIES",
        "CHECK_APPOINTMENT_AVAILABILITY",
        "REQUEST_APPOINTMENT",
        "CAPTURE_LEAD",
        "HANDOFF_TO_CLINIC",
    ),
)
